Source separation baseline for improvements

SebChw opened the pull request:
There are still some minor issues with Lightning logging - I don't see loss on tqdm loop. Other than that it seems to have similar training dynamics as training on `demucs` repository. 

closes #26 
closes #30

SebChw committed:
add: sourcesep baseline

SebChw committed:
add: sourcesep baseline

kordc commented on art/configs/datamodule/sdx_labelnoise.yaml:
why add an empty file?

SebChw replied:
this should be probably filled

kordc commented on art/configs/logger/neptune.yaml:
```diff
@@ -1,5 +1,5 @@
 _target_: lightning.pytorch.loggers.NeptuneLogger
 api_key: ${oc.env:NEPTUNE_API_TOKEN} # api key is loaded from environment variable
-project: skdbmk/wake-word-detection
+project: skdbmk/sourceseparation
```
maybe we can have one big Neptune project, where we can train several tasks at once? It'd just require adding a label to each run

SebChw replied:
to be discussed. For me many projects are better.

SebChw committed:
add: l1 measurement

SebChw committed:
Merge branch 'sebchw-relabeling' into sebchw-sourcesep-baseline

SebChw committed:
fix: make it train

trebacz626 committed:
missing requirements

trebacz626 committed:
fix dataloader

SebChw committed:
fix: utils

SebChw committed:
add: update datasets

SebChw committed:
Merge branch 'sebchw-sourcesep-baseline' of https://github.com/SebChw/audio-research-toolkit into sebchw-sourcesep-baseline

SebChw committed:
add: sanity check dataset

SebChw committed:
add: allow recursive init

SebChw committed:
add: zero loss blueprint

SebChw committed:
make sdr optional

SebChw committed:
allow L1 tracking

SebChw committed:
add: clean validation

trebacz626 committed:
precision update

SebChw committed:
allow load from disk

SebChw committed:
Merge branch 'sebchw-sourcesep-baseline' of https://github.com/SebChw/audio-research-toolkit into sebchw-sourcesep-baseline

SebChw committed:
increase num windows

SebChw committed:
fix: to short array

SebChw committed:
fix: plots

SebChw committed:
fix: y_axis

SebChw committed:
fix: plot memory leak

trebacz626 committed:
thresholding

trebacz626 committed:
merge thresholding

kordc committed:
add neptune api wrapper for getting the latest good run

trebacz626 committed:
fix requirement

trebacz626 committed:
fix requirement 2

SebChw committed:
fix: plotter

SebChw committed:
fix: optim direction

SebChw committed:
Merge branch 'sebchw-sourcesep-baseline' of https://github.com/SebChw/audio-research-toolkit into sebchw-sourcesep-baseline

SebChw committed:
fix: make collate a class

trebacz626 committed:
log log hist

SebChw committed:
fix: memory leak

SebChw committed:
Merge branch 'sebchw-sourcesep-baseline' of https://github.com/SebChw/audio-research-toolkit into sebchw-sourcesep-baseline

SebChw committed:
fix: thresholder

kordc committed:
make it cmd app

kordc committed:
Add feature enabling continuing training from a checkpoint in neptune.
This is done by adding a new parameter to the train.yaml file,
continue_training_id which is the id of the experiment from which the
checkpoint should be loaded. The checkpoint is loaded from the
neptune storage and the training is continued from there.
This is useful for example when the training was interrupted and
should be continued from the last checkpoint.
For now only the best checkpoint is supported

kordc committed:
wrap last commit

kordc committed:
disable cfg override, WIP
Current error:
neptune.internal.operation_processors.async_operation_processor][ERROR] - Error occurred during asynchronous operation processing: Timestamp must be non-decreasing for series attribute: monitoring/0e88a6af/stdout. Invalid point: 2023-04-15T11:24:25.140Z

SebChw committed:
Update tests.yml

kordc committed:
basic start without checkpoint

kordc committed:
add augmentations from demucs repo - in testing!

kordc committed:
don't continue

kordc committed:
fix functions

kordc committed:
revert the augmentations in litmodules

kordc committed:
final augmentations

SebChw committed:
add: mock logger

SebChw committed:
fix: subprocess spawning

SebChw committed:
fix: config uploading

SebChw committed:
fix: training continuation

SebChw committed:
fix: too fast checkpoints

SebChw commented:
After my latest commit, these are introduced changes/bug fixes:

* You can now use mocked logger with `python -m art.train logger=mock`
* Sanity check dataloader won't spawn process after every epoch. Just set persist_workers=True or n_workers=0 
* Now we upload raw `yaml` file as our configuration not dict.
* If you use sanity_check datamodule, please use neptune_no_check/mock logger then. Epochs are going very fast during sanity checking and big models won't be updated + we introduce a lot of network traffic
* Now training continuation works.

trebacz626 committed:
instrument threshold
